### PR TITLE
Enhance parser "orosync_cmapctl"

### LINKED ISSUE #2601: Enhance parser "corosync_cmapctl"
The clear schedule event way is different on RHEL 7 and RHEL 8.
On RHEL 7, it is 
```
corosync-cmapctl -d runtime.schedmiss.timestamp && corosync-cmapctl -d runtime.schedmiss.delay
```
On RHEL 8, it is 
```
corosync-cmapctl -C schedmiss
```
So I'll enhance the parser to run it with some dependency.

### PR DESCRIPTION
**Pull request by huali027**
* Fixes #2601
* The way of clearing schedule events data  is different on RHEL 7 and RHEL 8
* so run these commands with dependency

Signed-off-by: Huanhuan Li <huali@redhat.com>

### CODE REVIEW

File: insights/specs/default.py [hunk 1]
```diff
@@ -199,12 +199,17 @@ def is_ceph_monitor(broker):
...
             "/usr/sbin/corosync-cmapctl",
             '/usr/sbin/corosync-cmapctl -m stats "stats.schedmiss."',
             '/usr/sbin/corosync-cmapctl -C schedmiss'
-        ], "%s")
+        ], "%s", deps=[IsRhel8])
+    corosync_cmapctl_on_rhel7 = foreach_execute(
+        [
+            "/usr/sbin/corosync-cmapctl",
+            'corosync-cmapctl -d runtime.schedmiss.timestamp && corosync-cmapctl -d runtime.schedmiss.delay',
+        ], "%s", deps=[IsRhel7])
```
Comment (xiangce) [inline insights/specs/default.py:212]: Better do it in this way:
```suggestion
    @datasource([IsRhel7, IsRhel8])
    def corosync_cmapctl_cmd_list(broker):
        if broker.get(IsRhel7):
            return ["/usr/sbin/corosync-cmapctl",  '/usr/sbin/corosync-cmapctl -m stats "stats.schedmiss.", '/usr/sbin/corosync-cmapctl -C schedmiss']
        if broker.get(IsRhel8):
            return ["/usr/sbin/corosync-cmapctl",  'corosync-cmapctl -d runtime.schedmiss.timestamp", 'corosync-cmapctl -d runtime.schedmiss.delay']
        raise SkipComponent()

    corosync_cmapctl = foreach_execute(corosync_cmapctl_cmd_list, "%s")
```

File: insights/specs/default.py [hunk 2]
```diff
@@ -199,12 +199,17 @@ def is_ceph_monitor(broker):
...
         [
             "/usr/sbin/corosync-cmapctl",
             '/usr/sbin/corosync-cmapctl -m stats "stats.schedmiss."',
             '/usr/sbin/corosync-cmapctl -C schedmiss'
-        ], "%s")
+        ], "%s", deps=[IsRhel8])
+    corosync_cmapctl_on_rhel7 = foreach_execute(
+        [
+            "/usr/sbin/corosync-cmapctl",
+            'corosync-cmapctl -d runtime.schedmiss.timestamp && corosync-cmapctl -d runtime.schedmiss.delay',
```
Comment (xiangce) [inline insights/specs/default.py:211]: Better remove the "&&" and discrete the commands into two.

File: insights/parsers/corosync_cmapctl.py [hunk 3]
```diff
@@ -57,3 +53,8 @@ def parse_content(self, content):
             key, value = [item.strip() for item in line.split('=')]
             key_without_parenthese = key.split()[0]
             self[key_without_parenthese] = value
+
+
+@parser(Specs.corosync_cmapctl_on_rhel8)
```
Comment (xiangce) [inline insights/parsers/corosync_cmapctl.py:58]: By using the `@datasource` as above, you can implement it in one single Parser for both RHEL7 and RHEL8

File: insights/specs/default.py (hunk 1 above)
Comment (huali027) [reply insights/specs/default.py:212]: @xiangce, cool, I'll update it.

File: insights/specs/default.py (hunk 2 above)
Comment (huali027) [reply insights/specs/default.py:211]: @xiangce, sure, I'll update it, thanks.

**Pull request merged**